tutorial3_distributions/demo/main.js

```diff
@@ -1,133 +1,127 @@
 // CONSTANTS
-const width = window.innerWidth * 0.8;
-const height = window.innerHeight * 0.8;
-const margin = { top: 20, bottom: 40, left: 40, right: 40 };
-const radius = 3;
+const constants = {
+  width: window.innerWidth * 0.7,
+  height: window.innerHeight * 0.7,
+  margin: { top: 20, bottom: 40, left: 40, right: 40 },
+  radius: 3,
+}
 
 // APPLICATION STATE
-let data = [];
-let selectedState;
+let state = {
+  data: [],
+  selectedParty: "D"
+}
 
 // GLOBALS
-// initialize a variable `svg` -- this is empty before we assign something to it
+// these variables allow us to access anything we manipulate in init() but need access to in draw(). All these variables are empty before we assign something to it.
 let svg;
 let xScale;
 let yScale;
 
-// load in data
+// DATA LOAD
 d3.json("../../data/environmentRatings.json", d3.autoType).then(raw_data => {
   console.log("raw_data", raw_data);
-  data = raw_data;
+  state.data = raw_data;
   init();
 });
 
+// INIT FUNCTION
 function init() {
   // this will be run *once* when the data finishes loading in
   /** SCALES */
+
   xScale = d3
     .scaleLinear()
-    .domain(d3.extent(data, d => d.ideology_rating))
-    .range([margin.left, width - margin.right]);
+    .domain(d3.extent(state.data, d => d.ideology_rating))
+    .range([constants.margin.left, constants.width - constants.margin.right]);
 
   yScale = d3
     .scaleLinear()
-    .domain(d3.extent(data, d => d.environmental_rating))
-    .range([height - margin.bottom, margin.top]);
+    .domain(d3.extent(state.data, d => d.environmental_rating))
+    .range([constants.height - constants.margin.bottom, constants.margin.top]);
 
   /** AXES */
   const xAxis = d3.axisBottom(xScale);
   const yAxis = d3.axisLeft(yScale);
 
-  /**
-   * `Set` reference: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set
-   * pull out all the state values from the data create a `Set` out of them
-   * This is useful for creating a list of just *unique* values.
-   * The `[...Set]` syntax 'unpacks' the set into an array so that we can use it in our `data()` method.
-   * To see what this does, try the following in your browser console:
-   *    new Set([1, 2, 1, 4, 5, 6, 4, 6, 10])
-   * and:
-   *    [...new Set([1, 2, 1, 4, 5, 6, 4, 6, 10])]
-   */
-  const uniqueStates = [...new Set(data.map(d => d.state))];
-  // add dropdown for interaction
-  // select reference: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/select
```

Comment:
as much as I think this is extremely helpful, and I would want to reference this in class, I think you're right in that this is a long script. Once I realized we wouldn't need `selectedState`, I thought this could be sacrificed and just used `['D', 'R', 'Both']` for `selectedParty`... what do you think?